calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -181,6 +192,15 @@ func (w *PeerWatcher) WatchBGPPeers(t *tomb.Tomb) error {
 			default:
 				goto restart
 			}
+		case secret := <-w.secretWatcherEventChan:
+			/* BGP secret related event - update the SecretUpdated field of affected peers */
+			w.log.Debugf("secret update event received for %s", secret)
+			for _, peer := range state {
+				if peer.BGPPeerSpec.Password != nil && peer.BGPPeerSpec.Password.SecretKeyRef != nil && peer.BGPPeerSpec.Password.SecretKeyRef.Name == secret {
+					w.log.Debugf("SecretUpdated field set for peer=%s", peer.BGPPeerSpec.PeerIP)
+					peer.SecretUpdated = true
```

Comment:
Instead of punting the update of the password to the resync step, we can probably call `w.updateBGPPeer()` directly here for the peers that reference this Secret.

Reply:
Same, we can probably do this in a further refactoring